File: RRFFilesManager/Intake/PreliminaryInfo.vb
Public Class PreliminaryInfo

    Public Property IntakeForm As Intake

    Public Function Validate() As Boolean
        If Me.MatterTypeComboBox.Text = "" Or Me.StaffInterviewerComboBox.Text = "" Or Me.HowHearComboBox.Text = "" Or ResponsibleLawyerComboBox.Text = "" Or LawyerComboBox.Text = "" Or DateOfLossDateTimePicker.CustomFormat = " " Then
            If Me.MatterTypeComboBox.Text = "" Then MsgBox("Please select: Matter Type") : Return False
            If Me.StaffInterviewerComboBox.Text = "" Then MsgBox("Please select: Staff Interviewer") : Return False
            If Me.HowHearComboBox.Text = "" Then MsgBox("Please select: How did you hear about us?") : Return False
            If Me.ResponsibleLawyerComboBox.Text = "" Then MsgBox("Please select: Responsible Lawyer") : Return False
            If Me.LawyerComboBox.Text = "" Then MsgBox("Please select: File Lawyer") : Return False
            If DateOfLossDateTimePicker.CustomFormat = " " Then MsgBox("Please select: Date of Loss") : Return False
        Else
            If MatterSubTypeComboBox.Items.Count > 0 And MatterSubTypeComboBox.Text = "" Then
                MsgBox("Please select: Matter Sub Type") : Return False
            End If
        End If
        Return True
    End Function

    Public Sub getNewFileNumber(lastFileNumber As String)

        If LawyerComboBox.Text <> "" Then

            Dim dr() As System.Data.DataRow
            'Dim lastFileNumber As String = ActionLogDBDataSet.Tables("Intakes").Rows(0)("FileNumber")
            Dim lastFileNumberYear As Integer = lastFileNumber.Substring(0, 4)
            Dim thisYear As Integer = Year(Now())
            If thisYear = lastFileNumberYear Then
                Dim nextFileString As String = "000"
                Dim lastFile As String = lastFileNumber.Substring(6, 3)
                Dim nextFileNumber As Integer = lastFile + 1
                If Len(nextFileNumber.ToString) = 1 Then nextFileString = "00" & nextFileNumber
                If Len(nextFileNumber.ToString) = 2 Then nextFileString = "0" & nextFileNumber
                If Len(nextFileNumber.ToString) = 3 Then nextFileString = nextFileNumber

                dr = ActionLogDBDataSet.Tables("FileLawyer").Select("Lawyer='" & LawyerComboBox.Text & "'")
                If dr.Length > 0 Then
                    Dim nextFile As String = thisYear & dr(0)("NumberID").ToString() & nextFileString
                    Me.FileNumberTextBox.Text = nextFile

                End If
            Else
                dr = ActionLogDBDataSet.Tables("FileLawyer").Select("Lawyer='" & LawyerComboBox.Text & "'")
                If dr.Length > 0 Then
                    Dim nextFileString As String = "001"
                    Dim nextFile As String = thisYear & dr(0)("NumberID").ToString() & nextFileString
                    Me.FileNumberTextBox.Text = nextFile

                End If

            End If
        Else
            Me.FileNumberTextBox.Clear()
        End If
    End Sub





    Private Sub MatterSubTypeComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles MatterSubTypeComboBox.SelectedIndexChanged
        Dim matterSubType = ActionLogDBDataSet.MatterSubTypes.FirstOrDefault(Function(s) s.MatterSubType = MatterSubTypeComboBox.Text)
        If Not matterSubType Is Nothing And Not String.IsNullOrEmpty(matterSubType("StatutoryNotice").ToString()) Then
            StatutoryNoticeBox.Text = Format(DateAdd("d", 10, DateOfLossDateTimePicker.Value), "MMM-dd-yyyy")
        Else
            StatutoryNoticeBox.Text = ""
        End If
    End Sub

    Private Sub DateOfLossDateTimePicker_ValueChanged(Sender As Object, e As EventArgs) Handles DateOfLossDateTimePicker.ValueChanged
        DateOfLossDateTimePicker.CustomFormat = "MMM-dd-yyyy"
        Dim dr() As System.Data.DataRow
        dr = ActionLogDBDataSet.Tables("MatterSubTypes").Select("MatterSubType='" & MatterSubTypeComboBox.Text & "'")
        If dr.Length > 0 Then
            Dim statutoryNotice As String = dr(0)("StatutoryNotice").ToString()
            If statutoryNotice <> "" Then Me.StatutoryNoticeBox.Text = Format(DateAdd("d", 10, Me.DateOfLossDateTimePicker.Value), "MMM-dd-yyyy")
        End If
        If DateOfLossDateTimePicker.CustomFormat = "MMM-dd-yyyy" Then
            LimitationPeriodTextBox.Text = Format(Sender.Value.AddYears(2), "MMM-dd-yyyy")
        End If
    End Sub

    Private Sub LawyerComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles LawyerComboBox.SelectedIndexChanged
        Dim lastFileNumber As String = ActionLogDBDataSet.Intakes.FirstOrDefault()?("FileNumber")
        getNewFileNumber(lastFileNumber)
    End Sub

    Private Sub PreliminaryInfo_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.Initialize()
    End Sub

    Private Sub Initialize()
        Me.HearAboutUsTableAdapter.Fill(Me.ActionLogDBDataSet.HearAboutUs)
        Me.StaffInterviewerTableAdapter.Fill(Me.ActionLogDBDataSet.StaffInterviewer)
        Me.ResponsibleLawyerTableAdapter.Fill(Me.ActionLogDBDataSet.ResponsibleLawyer)
        Me.FileLawyerTableAdapter.Fill(Me.ActionLogDBDataSet.FileLawyer)
        Me.IntakesTableAdapter.Fill(Me.ActionLogDBDataSet.Intakes)
        Me.MatterTypeTableAdapter.Fill(ActionLogDBDataSet.MatterType)
        Me.MatterSubTypesTableAdapter.Fill(ActionLogDBDataSet.MatterSubTypes)
        Dim obj = IntakesBindingSource.AddNew()

        MatterSubTypeComboBox.Items.Clear()
        For Each row As DataRow In ActionLogDBDataSet.MatterSubTypes.Rows
            If row("Matter").ToString = "Motor Vehicle Accident" Then
                Me.MatterSubTypeComboBox.Items.Add(row("MatterSubType").ToString)
            End If
        Next
        Me.LawyerComboBox.SelectedIndex = -1
        Dim lastFileNumber As String = ActionLogDBDataSet.Tables("Intakes").Rows(0)("FileNumber")
        getNewFileNumber(lastFileNumber)
        DateOfLossDateTimePicker.Format = DateTimePickerFormat.Custom
        DateOfLossDateTimePicker.CustomFormat = " "
        Me.MatterTypeComboBox.SelectedValue = vbNull
        Me.LimitationPeriodTextBox.Text = ""
    End Sub

    Public Sub OnNext()
        IntakeForm.IntakesBindingSource.EndEdit()
        IntakeForm.IntakesTableAdapter.Update(ActionLogDBDataSet.Intakes)
    End Sub

    Private Sub MatterTypeComboBox_SelectedIndexChanged(sender As Object, e As EventArgs) Handles MatterTypeComboBox.SelectedIndexChanged

    End Sub
End Class

File: RRFFilesManager/Intake/PreliminaryInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace RRFFilesManager.Intake
{
    public partial class PreliminaryInfo : UserControl
    {
        public PreliminaryInfo()
        {
            InitializeComponent();
        }

        private static PreliminaryInfo instance;
        public static PreliminaryInfo Instance => instance ?? (instance = new PreliminaryInfo());

        public new bool Validate()
        {
            if (string.IsNullOrEmpty(MatterTypeComboBox.Text) | string.IsNullOrEmpty(StaffInterviewerComboBox.Text) | string.IsNullOrEmpty(HowHearComboBox.Text) | string.IsNullOrEmpty(ResponsibleLawyerComboBox.Text) | string.IsNullOrEmpty(LawyerComboBox.Text) | DateOfLossDateTimePicker.CustomFormat == " ")
            {
                if (string.IsNullOrEmpty(MatterTypeComboBox.Text))
                {
                    MessageBox.Show("Please select: Matter Type");
                    return false;
                }

                if (string.IsNullOrEmpty(StaffInterviewerComboBox.Text))
                {
                    MessageBox.Show("Please select: Staff Interviewer");
                    return false;
                }

                if (string.IsNullOrEmpty(HowHearComboBox.Text))
                {
                    MessageBox.Show("Please select: How did you hear about us?");
                    return false;
                }

                if (string.IsNullOrEmpty(ResponsibleLawyerComboBox.Text))
                {
                    MessageBox.Show("Please select: Responsible Lawyer");
                    return false;
                }

                if (string.IsNullOrEmpty(LawyerComboBox.Text))
                {
                    MessageBox.Show("Please select: File Lawyer");
                    return false;
                }

                if (DateOfLossDateTimePicker.CustomFormat == " ")
                {
                    MessageBox.Show("Please select: Date of Loss");
                    return false;
                }
            }
            else if (MatterSubTypeComboBox.Items.Count > 0 & string.IsNullOrEmpty(MatterSubTypeComboBox.Text))
            {
                MessageBox.Show("Please select: Matter Sub Type");
                return false;
            }

            return true;
        }

        public void getNewFileNumber(string lastFileNumber)
        {
            if (!string.IsNullOrEmpty(LawyerComboBox.Text))
            {
                DataRow[] dr;
                // Dim lastFileNumber As String = ActionLogDBDataSet.Tables("Intakes").Rows(0)("FileNumber")
                int lastFileNumberYear = Convert.ToInt32(lastFileNumber.Substring(0, 4));
                int thisYear = DateTime.Now.Year;
                if (thisYear == lastFileNumberYear)
                {
                    string nextFileString = "000";
                    string lastFile = lastFileNumber.Substring(6, 3);
                    int nextFileNumber = (int)(Convert.ToDouble(lastFile) + 1d);
                    if (nextFileNumber.ToString().Length == 1)
                        nextFileString = "00" + nextFileNumber;
                    if (nextFileNumber.ToString().Length == 2)
                        nextFileString = "0" + nextFileNumber;
                    if (nextFileNumber.ToString().Length == 3)
                        nextFileString = nextFileNumber.ToString();
                    dr = ActionLogDBDataSet.Tables["FileLawyer"].Select("Lawyer='" + LawyerComboBox.Text + "'");
                    if (dr.Length > 0)
                    {
                        string nextFile = thisYear + dr[0]["NumberID"].ToString() + nextFileString;
                        FileNumberTextBox.Text = nextFile;
                    }
                }
                else
                {
                    dr = ActionLogDBDataSet.Tables["FileLawyer"].Select("Lawyer='" + LawyerComboBox.Text + "'");
                    if (dr.Length > 0)
                    {
                        string nextFileString = "001";
                        string nextFile = thisYear + dr[0]["NumberID"].ToString() + nextFileString;
                        FileNumberTextBox.Text = nextFile;
                    }
                }
            }
            else
            {
                FileNumberTextBox.Clear();
            }
        }

        private void MatterSubTypeComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            var matterSubType = ActionLogDBDataSet.MatterSubTypes.FirstOrDefault(s => (s.MatterSubType ?? "") == (MatterSubTypeComboBox.Text ?? ""));
            if (matterSubType is object & !string.IsNullOrEmpty(matterSubType["StatutoryNotice"].ToString()))
            {
                StatutoryNoticeBox.Text = DateOfLossDateTimePicker.Value.AddDays(10).ToString("MMM-dd-yyyy");
            }
            else
            {
                StatutoryNoticeBox.Text = "";
            }
        }

        private void DateOfLossDateTimePicker_ValueChanged(object Sender, EventArgs e)
        {
            DateOfLossDateTimePicker.CustomFormat = "MMM-dd-yyyy";
            DataRow[] dr;
            dr = ActionLogDBDataSet.Tables["MatterSubTypes"].Select("MatterSubType='" + MatterSubTypeComboBox.Text + "'");
            if (dr.Length > 0)
            {
                string statutoryNotice = dr[0]["StatutoryNotice"].ToString();
                if (!string.IsNullOrEmpty(statutoryNotice))
                    StatutoryNoticeBox.Text = DateOfLossDateTimePicker.Value.AddDays(10).ToString("MMM-dd-yyyy");
            }

            if (DateOfLossDateTimePicker.CustomFormat == "MMM-dd-yyyy")
            {
                LimitationPeriodTextBox.Text = DateOfLossDateTimePicker.Value.AddYears(2).ToString("MMM-dd-yyyy");
            }
        }

        private void LawyerComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {

            var lastFileNumber = ActionLogDBDataSet.Intakes.FirstOrDefault().FileNumber.ToString();
            getNewFileNumber(lastFileNumber);
        }

        private void PreliminaryInfo_Load(object sender, EventArgs e)
        {
            Initialize();
        }

        private void Initialize()
        {
            HearAboutUsTableAdapter.Fill(ActionLogDBDataSet.HearAboutUs);
            StaffInterviewerTableAdapter.Fill(ActionLogDBDataSet.StaffInterviewer);
            ResponsibleLawyerTableAdapter.Fill(ActionLogDBDataSet.ResponsibleLawyer);
            FileLawyerTableAdapter.Fill(ActionLogDBDataSet.FileLawyer);
            IntakesTableAdapter.Fill(ActionLogDBDataSet.Intakes);
            MatterTypeTableAdapter.Fill(ActionLogDBDataSet.MatterType);
            MatterSubTypesTableAdapter.Fill(ActionLogDBDataSet.MatterSubTypes);
            //var obj = IntakesBindingSource.AddNew();
            MatterSubTypeComboBox.Items.Clear();
            foreach (DataRow row in ActionLogDBDataSet.MatterSubTypes.Rows)
            {
                if (row["Matter"].ToString() == "Motor Vehicle Accident")
                {
                    MatterSubTypeComboBox.Items.Add(row["MatterSubType"].ToString());
                }
            }

            LawyerComboBox.SelectedIndex = -1;
            string lastFileNumber = ActionLogDBDataSet.Tables["Intakes"].Rows[0]["FileNumber"].ToString();
            getNewFileNumber(lastFileNumber);
            DateOfLossDateTimePicker.Format = DateTimePickerFormat.Custom;
            DateOfLossDateTimePicker.CustomFormat = " ";
            LimitationPeriodTextBox.Text = "";
        }

        public void OnNext()
        {
            //IntakeForm.IntakesBindingSource.EndEdit();
            //IntakeForm.IntakesTableAdapter.Update(ActionLogDBDataSet.Intakes);
        }

        private void MatterTypeComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            var Provinces = Program.DataContext.Provinces.ToList();
        }
    }
}
